californium-core/src/main/java/org/eclipse/californium/core/network/stack/TCPBlockwiseLayer.java

```diff
@@ -0,0 +1,394 @@
+package org.eclipse.californium.core.network.stack;
+
+import java.nio.ByteBuffer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.core.coap.BlockOption;
+import org.eclipse.californium.core.coap.OptionSet;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.coap.CoAP.ResponseCode;
+import org.eclipse.californium.core.network.Exchange;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.core.network.config.NetworkConfig.Keys;
+
+/**
+ * Blockwise layer for TCP. Extended mainly for BERT option. Capable of handling
+ * incoming blocks with SZX = 7.
+ * 
+ * @author Pratheek Rai.
+ *
+ */
+public class TCPBlockwiseLayer extends BlockwiseLayer {
+
+	/**
+	 * logger.
+	 */
+	private static final Logger LOGGER = Logger.getLogger(TCPBlockwiseLayer.class.getName());
+
+	/**
+	 * BERT SZX constant.
+	 */
+	private final static int BERT_SZX = 7;
+	
+	/**
+	 * Internal block size for BERT. i.e internally BERT is handled similar to
+	 * SZX = 6.
+	 */
+	private final static int BERT_INT_BLOCK_SIZE = 1024;
+
+	/**
+	 * The number of 1024 Bytes packets to be sent as a block. Size of 1 BERT
+	 * block = bertStepSize * 1024 Bytes
+	 */
+	private int bertStepSize;
+
+	/**
+	 * Flag for BERT option.
+	 */
+	private boolean isBertEnabled;
```

Comment:
I prefer having this flag because I want to use TcpBlockwiseLayer in CoapTcpStack. I think it would help to receive incoming BERT messages even  when the BERT sending option is disabled. This could be useful until we have the Messaging implemented.